### DISCUSSION TITLE
Two Tables One Wheel Help

### DISCUSSION THREAD

**Discussion by topper-2k** [PinballY/PinballY-Addons-and-Examples#3] [Q&A] [answered]
I'm thrilled to see the "Two Tables One Wheel" script, but I'm having a little trouble getting it to work. 

I got the script downloaded and included in my main.js. And I created the isAlternateVersion category. I started with Addams Family, which I have a couple versions of and named them in PBY via Game Setup as follows. 

VPX: Addams Family (Midway 1992)
Future Pinball: Addams Family (Future Pinball) (Midway 1992)

Then I assigned the FP version to the isAlternateVersion category.

After completing those steps, when I go into the operator menu, I see the option for "Show Alternate Versions." And checking/unchecking that option makes the FP version of Addams Family show/hide in my main wheel. So I think the script is included and working. 

But, with "Show Alternate Versions" unchecked, when I go to the Addams Family wheel icon and push the start button, I do not get the additional menu options you show. I only get a single "Play" option at the top. There is no "Play Future Pinball Version" listed. 

Am I missing a critical step somewhere?

**Comment by stephenhouser** [comment]
The names look right. Check the pinbally.log file. See if there is some error showing up there.

**Reply by stephenhouser** [reply]
Another thing to check is in your database XML files to ensure the `<description>` tags are:

* `Addams Family (Midway 1992)`
* `Addams Family (Future Pinball) (Midway 1992)`

Maybe some extra spaces in there or something?

**Comment by topper-2k** [comment]
Looking at the XML file helped. 

It appears that game name attribute has to follow the naming convention you outline in the read me. Editing the PinballY game name via the operator menu only updates the description element. But that doesn't seem to affect the script.

I *think* the game name attribute is linked to the table file name and doesn't have anything to do with the PBY operator menu. 

Changing the file name of the tables and manually editing the XML seems to be the only  way to get this to work. 

Is that normal and I misread the instructions? Or is something wonky in my PBY setup?

**Reply by stephenhouser** [reply]
My intent was to have the name you set in the operator menu be what it matches on. I'll fix it if it's not! Can you send me your XML file that was not working?

**Reply by topper-2k** [reply]
Sure. I need to undo some of what I've done for testing so you can see how it was structured by PBY before I manually edited it. On further investigation, I don't think the game name attribute in the xml is related to the file name.

**Comment by topper-2k** [comment]
Here's what my PBY table settings look like.
![Addams-FP](https://user-images.githubusercontent.com/43299569/109395944-22c00280-78fd-11eb-8104-76838e232700.JPG)
![Addams-VPX](https://user-images.githubusercontent.com/43299569/109395945-22c00280-78fd-11eb-9c54-0bf85c104625.JPG)

**Comment by topper-2k** [comment]
Here's the XML from the two database files related to these games.

Visual Pinball X.xml
`<game name="Addams Family Special Collectors Edition, The (Bally 1994)">
		<ipdbid>21</ipdbid>
		<description>Addams Family (Midway 1992) (Midway 1992)</description>
		<type>SS</type>
		<rom></rom>
		<manufacturer>Midway</manufacturer>
		<year>1992</year>
		<rating>0</rating>
		<enabled>True</enabled>
	</game>`

Future Pinball.xml
	`<game name="The Addams Family ULTIMATE Pro 1.05">
		<ipdbid>20</ipdbid>
		<description>Addams Family (Future Pinball) (Midway 1992) (Midway 1992)</description>
		<type>SS</type>
		<rom></rom>
		<manufacturer>Midway</manufacturer>
		<year>1992</year>
		<rating>0</rating>
	</game>`

**Reply by topper-2k** [reply]
Hold on, I might be totally wrong here. I'm trying to reproduce my previous steps and it's not working the same way now.

Now it's working as described in the ReadMe with one exception. I left (Midway 1992) out of the game name in PBY game setup for both tables. So VPX is "Addams Family" and FP is "Addams Family (Future Pinball)"

I think I sent you astray about editing the xml game name attribute. That's not having any effect now and I'm not sure why I thought it did the first time. I wonder if I changed both the game name attribute and the description that first time through.

**Comment by topper-2k** [comment, marked as answer]
Here are all new screenshots. 

Once I dropped the (Midway 1992) from the PBY game details screen everything is working as expected. Sorry about the wild goose chase on the game name attribute in the xml. Chalk it up to a testing error.

This is fantastic. If there's a way to modify the menu order (so the play options are above the apply meta filter) please point me in the right direction. But that's a really minor thing. I'm thrilled with this and am off to rename some tables!

![Addams-FP-working](https://user-images.githubusercontent.com/43299569/109396489-d7f3ba00-78ff-11eb-9a07-1f45479c046f.JPG)
![Addams-VPX-working](https://user-images.githubusercontent.com/43299569/109396490-d7f3ba00-78ff-11eb-97a7-e418161934cc.JPG)
![wheel-working](https://user-images.githubusercontent.com/43299569/109396491-d7f3ba00-78ff-11eb-85ef-ad232dc3aa83.JPG)

**Reply by stephenhouser** [reply]
Set metafilter must be another plugin/script you have running. It is likely getting to the menu after this script and adding itself right after play. Check your main.js file, maybe change the order of your `import` statements.

**Reply by topper-2k** [reply]
Yep! It's one of the other scripts in this repository. I was able to move it by changing this line in its script:

ev.addMenuItem({ after: command.PlayGame }

to 

ev.addMenuItem({ after: command.RateGame }

Many thanks for all your help, both to me personally to get this up and running. And also for getting this repository setup. It's going to be super useful.
